src/requests/features.jl

```diff
@@ -236,16 +210,14 @@ function textDocument_range_formatting_request(params::DocumentRangeFormattingPa
         )
     end
 
-    if longest_prefix !== nothing && !isempty(longest_prefix)
-        io = IOBuffer()
-        for line in eachline(IOBuffer(newcontent), keep=true)
-            print(io, longest_prefix, line)
-        end
-        newcontent = String(take!(io))
+    range_formatted = match(Regex("$startmark\n((?s).*\n)\\s*$stopmark"), text_formatted)
+    if isnothing(range_formatted)
+        newcontent = oldcontent
+    else
+        newcontent = replace(text_marked, Regex("$startmark\n(?s).*$stopmark\n") => range_formatted[1])
     end
-
-    lsedits = TextEdit[TextEdit(Range(l1, c1, l2, c2), newcontent)]
-
+    end_l, end_c = get_position_from_offset(doc, sizeof(get_text(doc))) # AUDIT: OK
+    lsedits = TextEdit[TextEdit(Range(0, 0, end_l, end_c), newcontent)]
     return lsedits
```

Comment:
I think this can just be 
```
    range_formatted = match(Regex("$startmark\n((?s).*)\n\\s*$stopmark"), text_formatted)
    if isnothing(range_formatted)
        return TextEdit[]
    end
    return TextEdit[TextEdit(Range(params.range.start.line, 0, params.range.stop.line, 99999999), range_formatted[1])]
```
so we don't replace the whole text document. `99999999` should probably be the actual line length though :)

Reply:
Ahh yes. This is better. By the way, I was thinking that perhaps it would be nice to avoid editing the file if the formatting makes no change to the text in the range. This would avoid marking the file as dirty when no change was made.